### PR TITLE
bugfixes: small tidy ups and bug fixes

### PR DESCRIPTION
**Pull request by msf**

### CODE REVIEW

File: client/duneapi/client.go [hunk 1]
```diff
@@ -111,24 +106,29 @@ func (c *client) sendRequest(ctx context.Context, request BlockchainIngestReques
...
 	c.log.Debug("Sending request", "url", url)
 	req, err := retryablehttp.NewRequestWithContext(ctx, "POST", url, bytes.NewReader(request.Payload))
 	if err != nil {
 		return err
 	}
-	req.Header.Set("Content-Type", request.ContentType)
+	if request.ContentEncoding != "" {
+		req.Header.Set("Content-Encoding", request.ContentEncoding)
+	}
+	req.Header.Set("Content-Type", "application/x-ndjson")
```
Comment (helanto) [inline client/duneapi/client.go:131]: Is it going to be always JSON ? Or JSON just for now ?

**Review by helanto — APPROVED** [review]
💪

File: ingester/mainloop.go [hunk 2]
```diff
@@ -56,29 +56,33 @@ func (i *ingester) ConsumeBlocks(
...
+	waitForBlock := func(ctx context.Context, blockNumber, latestBlockNumber int64) int64 {
 		for blockNumber > latestBlockNumber {
 			select {
 			case <-ctx.Done():
 				return latestBlockNumber
-			default:
+			case <-time.After(i.cfg.PollInterval):
 			}
-			i.log.Info(fmt.Sprintf("Waiting %v for block to be available..", i.cfg.PollInterval),
+			i.log.Debug(fmt.Sprintf("Waiting %v for block to be available..", i.cfg.PollInterval),
```
Comment (vegarsti) [inline ingester/mainloop.go:66]: -> waited? 😄

File: ingester/mainloop.go [hunk 3]
```diff
@@ -56,29 +56,33 @@ func (i *ingester) ConsumeBlocks(
...
-		latestBlockNumber = waitForBlock(blockNumber, latestBlockNumber)
+		latestBlockNumber = waitForBlock(ctx, blockNumber, latestBlockNumber)
 		startTime := time.Now()
 
 		block, err := i.node.BlockByNumber(ctx, blockNumber)
 		if err != nil {
+			if errors.Is(err, context.Canceled) {
+				i.log.Info("Context canceled, stopping..")
+				return err
+			}
```
Comment (vegarsti) [inline ingester/mainloop.go:84]: ah nice, this fixed the racy test perhaps?

**Review by vegarsti — APPROVED** [review]
Great!

File: client/duneapi/client.go (hunk 1 above)
Comment (msf) [reply client/duneapi/client.go:131]: for now, but great question

**Pull request merged**